File: code/drasil-code/test/golden/PatternTestOO/csharp/PatternTest.cs
using System;
using System.Collections.Generic;

public class PatternTest {
    
    public static void Main(string[] args) {
        int n;
        
        Console.WriteLine("myStrat");
        n = 3;
        
        Observer obs1 = new Observer();
        Observer obs2 = new Observer();
        
        List<Observer> observerList = new List<Observer> {obs1};
        observerList.Insert(observerList.Count, obs2);
        for (int observerIndex = 0; observerIndex < observerList.Count; observerIndex++) {
            observerList[observerIndex].printNum();
        }
        
        obs1.setX(10);
        Console.Write(obs1.getX());
    }
}

File: code/drasil-code/test/golden/PatternTestOO/cpp/PatternTest.cpp
#include <iostream>
#include <iterator>
#include <string>
#include <vector>

#include "Observer.hpp"

using std::string;
using std::vector;

int main(int argc, const char *argv[]) {
    int n;
    
    std::cout << "myStrat" << std::endl;
    n = 3;
    
    Observer obs1 = Observer();
    Observer obs2 = Observer();
    
    vector<Observer> observerList{obs1};
    observerList.insert(observerList.begin() + (int)(observerList.size()), obs2);
    for (int observerIndex = 0; observerIndex < (int)(observerList.size()); observerIndex++) {
        observerList.at(observerIndex).printNum();
    }
    
    obs1.setX(10);
    std::cout << obs1.getX();
    
    return 0;
}
